ui/src/app/cohort-review/annotation-definition-modals/annotation-definition-modals.component.tsx

```diff
@@ -146,28 +151,33 @@ export const EditAnnotationDefinitionsModal = withUrlParams()(class extends Reac
     setAnnotationDefinitions: Function,
     urlParams: any
   },
-  {editId: number, editValue: string, busy: boolean}
+  {editId: number, editValue: string, busy: boolean, deleteId: number, deleteError: boolean,
+    renameError: boolean}
 > {
   constructor(props) {
     super(props);
-    this.state = {editId: undefined, editValue: '', busy: false};
+    this.state = {editId: undefined, editValue: '', busy: false, deleteId: undefined,
+      deleteError: false, renameError: false};
   }
 
   async delete(id) {
     try {
       const {
-        annotationDefinitions, setAnnotationDefinitions,
+        annotationDefinitions, onClose, setAnnotationDefinitions,
         urlParams: {ns, wsid, cid}
       } = this.props;
       this.setState({busy: true});
       await cohortAnnotationDefinitionApi().deleteCohortAnnotationDefinition(ns, wsid, cid, id);
       setAnnotationDefinitions(
         fp.remove({cohortAnnotationDefinitionId: id}, annotationDefinitions)
       );
+      onClose(true);
     } catch (error) {
       console.error(error);
+      this.setState({deleteError: true});
+      setTimeout(() => this.setState({deleteError: false}), 5000);
     } finally {
-      this.setState({busy: false});
+      this.setState({busy: false, deleteId: undefined});
```

Comment:
Actually, this one won't give a compile error since it's not in the constructor, but we do need to set it back to its initial state (`undefined`) to show that the delete completed.